// @progress/kendo-vue-dialogs/dist/es/WindowTitlebar.js
// @ts-ignore
import { getTemplate } from '@progress/kendo-vue-common';
import * as Vue from 'vue';
var allVue = Vue;
var gh = allVue.h;
import { windowStage } from './StageEnum';
/**
 * Represents the default `WindowTitleBar` component.
 */

var WindowTitleBar = {
  props: {
    id: String,
    stage: String,
    closeButton: [String, Function, Object],
    minimizeButton: [String, Function, Object],
    maximizeButton: [String, Function, Object],
    restoreButton: [String, Function, Object],
    title: String,
    titleRender: [String, Function, Object]
  },
  // @ts-ignore
  setup: !gh ? undefined : function () {
    var v3 = !!gh;
    return {
      v3: v3
    };
  },
  methods: {
    onDoubleClick: function onDoubleClick(e) {
      this.$emit('doubleclick', e);
    },
    onMinimizeClick: function onMinimizeClick(e) {
      this.$emit('minimizeclick', e);
    },
    onFullScreenClick: function onFullScreenClick(e) {
      this.$emit('fullscreenclick', e);
    },
    onRestoreClick: function onRestoreClick(e) {
      this.$emit('restoreclick', e);
    },
    onCloseClick: function onCloseClick(e) {
      this.$emit('closeclick', e);
    }
  },
  // @ts-ignore
  render: function render(createElement) {
    var h = gh || createElement;
    var props = this.$props;
    var stage = props.stage,
        title = props.title,
        titleRender = props.titleRender,
        minimizeButton = props.minimizeButton,
        maximizeButton = props.maximizeButton,
        restoreButton = props.restoreButton,
        closeButton = props.closeButton;
    var titleElement = getTemplate.call(this, {
      h: h,
      template: titleRender,
      defaultRendering: title
    });
    var minimizeButtonDefault = h("button", {
      "class": "k-button-icon k-button k-flat",
      onClick: this.onMinimizeClick,
      on: this.v3 ? undefined : {
        "click": this.onMinimizeClick
      }
    }, [h("span", {
      "class": "k-icon k-i-window-minimize",
      role: "presentation",
      attrs: this.v3 ? undefined : {
        role: "presentation"
      }
    })]);
    var minimizeButtonRender = getTemplate.call(this, {
      h: h,
      template: minimizeButton,
      defaultRendering: minimizeButtonDefault
    });
    var maximizeButtonDefault = h("button", {
      "class": "k-button-icon k-button k-flat",
      onClick: this.onFullScreenClick,
      on: this.v3 ? undefined : {
        "click": this.onFullScreenClick
      }
    }, [h("span", {
      "class": "k-icon k-i-window-maximize",
      role: "presentation",
      attrs: this.v3 ? undefined : {
        role: "presentation"
      }
    })]);
    var maximizeButtonRender = getTemplate.call(this, {
      h: h,
      template: maximizeButton,
      defaultRendering: maximizeButtonDefault
    });
    var restoreButtonDefault = h("button", {
      "class": "k-button-icon k-button k-flat",
      onClick: this.onRestoreClick,
      on: this.v3 ? undefined : {
        "click": this.onRestoreClick
      }
    }, [h("span", {
      "class": "k-icon k-i-window-restore",
      role: "presentation",
      attrs: this.v3 ? undefined : {
        role: "presentation"
      }
    })]);
    var restoreButtonRender = getTemplate.call(this, {
      h: h,
      template: restoreButton,
      defaultRendering: restoreButtonDefault
    });
    var closeButtonDefault = h("button", {
      "class": "k-button-icon k-button k-flat",
      onClick: this.onCloseClick,
      on: this.v3 ? undefined : {
        "click": this.onCloseClick
      }
    }, [h("span", {
      "class": "k-icon k-i-close",
      role: "presentation",
      attrs: this.v3 ? undefined : {
        role: "presentation"
      }
    })]);
    var closeButtonRender = getTemplate.call(this, {
      h: h,
      template: closeButton,
      defaultRendering: closeButtonDefault
    });
    return h("div", {
      "class": "k-window-titlebar k-dialog-titlebar",
      style: {
        touchAction: 'none'
      },
      onDblclick: this.onDoubleClick,
      on: this.v3 ? undefined : {
        "dblclick": this.onDoubleClick
      }
    }, [h("div", {
      "class": "k-window-title"
    }, [titleElement]), h("div", {
      "class": "k-window-actions"
    }, [stage === windowStage.DEFAULT && minimizeButtonRender, stage === windowStage.DEFAULT && maximizeButtonRender, stage !== windowStage.DEFAULT && restoreButtonRender]), closeButtonRender]);
  }
};
export { WindowTitleBar };
